scipy/interpolate/tests/test_interpolate.py

```diff
@@ -2523,21 +2837,50 @@ def _sample_2d_data(self):
                       [1, 2, 2, 2, 1], [1, 2, 1, 2, 1]])
         return x, y, z
 
-    def test_spline_2d(self):
-        x, y, z = self._sample_2d_data()
-        lut = RectBivariateSpline(x, y, z)
+    def _get_sample_4d_large(self):
+        def f(x, y, z, w):
+            return x**2 + y**2 + z**2 + w**2
+        X = np.linspace(-10, 10, 6)
+        Y = np.linspace(-10, 10, 7)
+        Z = np.linspace(-10, 10, 6)
+        W = np.linspace(-10, 10, 8)
+        points = [X, Y, Z, W]
+        values = f(*np.meshgrid(*points, indexing='ij'))
+        return points, values
+
+    def _sample_2d_data_smooth(self):
+        # test problem with enough points for smooth spline fits
+        def f(u, v):
+            return u * np.cos(u * v) + v * np.sin(u * v)
+
+        def df(u, v):
+            return (-u * v * np.sin(u * v) + v**2 * np.cos(u * v) +
+                    np.cos(u * v),
+                    -u**2 * np.sin(u * v) + u * v * np.cos(u * v) +
+                    np.sin(u * v))
+        u = np.linspace(0, 3, 40)
+        v = np.linspace(0, 3, 40)
+        points = [u, v]
+        values = f(*np.meshgrid(*points, indexing='ij'))
+        return points, values, f, df
 
-        xi = np.array([[1, 2.3, 5.3, 0.5, 3.3, 1.2, 3],
-                       [1, 3.3, 1.2, 4.0, 5.0, 1.0, 3]]).T
-        assert_array_almost_equal(interpn((x, y), z, xi, method="splinef2d"),
+    def test_spline_2d(self):
+        points, values, f, df = self._sample_2d_data_smooth()
+        x, y = points
+        lut = RectBivariateSpline(x, y, values)
+        np.random.seed(2)
+        xi = np.random.uniform(0, 3, (5, 2))
+        assert_array_almost_equal(interpn((x, y), values, xi,
+                                          method="splinef2d"),
                                   lut.ev(xi[:, 0], xi[:, 1]))
 
     def test_list_input(self):
-        x, y, z = self._sample_2d_data()
-        xi = np.array([[1, 2.3, 5.3, 0.5, 3.3, 1.2, 3],
-                       [1, 3.3, 1.2, 4.0, 5.0, 1.0, 3]]).T
+        points, z, f, df = self._sample_2d_data_smooth()
+        x, y = points
+        np.random.seed(3)
```

Comment:
A very minor nitpick, optional: a recommended idiom is to create a `RandomState` instance instead of seeding the global generator. We do not really follow this, and there is no need for a code churn on this account alone, but in new code maybe better follow the recommendation.